src/lib.rs

```diff
@@ -78,95 +80,38 @@ impl<R> Volatile<R> {
     /// volatile.write(1);
     /// assert_eq!(volatile.read(), 1);
     /// ```
-    pub const fn new(reference: R) -> Volatile<R> {
+    pub unsafe fn new<A>(reference: &UnsafeCell<T>, access: A) -> Volatile<'_, T, A> {
```

Comment:
I think this is safe.

Reply:
Thanks for taking a look! This function could be safe, but I think we would need to make all read/write functions unsafe then because e.g. race conditions could occur if the `UnsafeCell` is also used in other threads.